Controller/RoleController.php

```diff
@@ -0,0 +1,183 @@
+            'can_delete' => $this->isGranted(new Attribute('role', 'delete')),
+        ]);
+    }
+
+    /**
+     * Creates a role.
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function createRoleAction()
+    {
+        try {
+            $roleCreateStruct = new RoleCreateStruct([
+                'identifier' => '__new__' . md5(microtime(true)),
+            ]);
+            $role = $this->roleService->createRole($roleCreateStruct);
+        } catch (UnauthorizedException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+        } catch (InvalidArgumentException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct? Name is already existing, or limitation errors
+        } catch (LimitationValidationException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct? Invalid policy limitation.
+        }
+
+        return $this->redirectToRoute(
+            'admin_roleUpdate',
+            ['roleId' => $role->id]
+        );
+    }
+
+    /**
+     * Updates a role.
+     *
+     * @param int $roleId Role ID
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function updateRoleAction(Request $request, $roleId)
+    {
+        try {
+            $role = $this->roleService->loadRole($roleId);
+        } catch (NotFoundException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct?
+        }
+
+        $roleData = (new RoleMapper())->mapToFormData($role);
+        $form = $this->createForm(
+            'ezrepoforms_role_update',
+            $roleData
+        );
+        $actionUrl = $this->generateUrl(
+            'admin_roleUpdate',
+            ['roleId' => $roleId]
+        );
+
+        // Synchronize form and data.
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            try {
+                $this->actionDispatcher->dispatchFormAction(
+                    $form,
+                    $roleData,
+                    $form->getClickedButton()->getName()
+                );
+            } catch (InvalidArgumentException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied'); // TODO: is this correct? Name is already existing.
+            } catch (UnauthorizedException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+            }
+
+            if ($response = $this->actionDispatcher->getResponse()) {
+                return $response;
+            }
+
+            return $this->redirect($actionUrl);
+        }
+
+        return $this->render('eZPlatformUIBundle:Role:update_role.html.twig', [
+            'form' => $form->createView(),
+            'action_url' => $actionUrl,
+            'role' => $role,
```

Comment:
Maybe also pass a "is_new" bool flag so template can empty name input and show "New Role" in heading. 
Can detect this with strpos('__new__') === 0 on name.